tox.ini

```diff
@@ -123,3 +123,8 @@ commands =
 [testenv:smoke]
 deps = -r{toxinidir}/requirements-test.txt
 commands = pytest tests/smoke {posargs}
+
+[testenv:pre-commit]
+skip_install = true
+deps = -r requirements-precommit.txt
+commands = pre-commit run --all-files --show-diff-on-failure
```

Comment:
If I understand correctly - this, together with `pre-commit` in the `envlist`, will actually still run duplicate checks when you run a default `tox`. I'd either take them out from the envlist, or specifically define which pre-commit hook ids we want to run. Otherwise, LGTM! :)

Reply:
Correct. It will run duplicate checks. I would prefer to leave it like that for the time being until @JohnVillalovos provides his feedback on his user expriance. Does that make sense?

Reply:
If it doesn't add too much runtime I don't mind and sounds ok to me if we deduplicate them in the end, will let @JohnVillalovos chime in as I joined in a bit late in this thread :)